### PR TITLE
feat: made Entire Radio Buttons Clickable, fixes to build.gradle files

### PR DESCRIPTION
**Pull request by M4dhav**
Made changes as requested in #13

### CODE REVIEW

**Comment by guyluz11** [comment]
Please change pr to dev branch, and preferably next time split unrelated fixes to a new pr (the build.gradle files).

Can you elaborate on the change to the build.gradle?, it run on my emulator so I am wondering why there was need for a change

**Comment by M4dhav** [comment]
Okay, I will fork from dev branch next time onwards. The code was not running on my emulator and was showing repeated gradle build errors. The changes I made were copied directly from the gradles of my previous projects. The changes include specifying a version of Kotlin to compile the code, and adding sources for gradle plugins. Hopefully these changes will make sure the project runs on a wider host of devices so future contributors do not have issues running it.

File: lib/presentation/molecules/study_type_selection_molecule.dart [hunk 1]
```diff
@@ -40,23 +40,29 @@ class _StudyTypeSelectionMoleculeState
   Widget build(BuildContext context) {
     return Column(
       children: [
-        ListTileAtom(
-          StudyType.analytically.previewName,
-          Radio<StudyType>(
-            value: StudyType.analytically,
-            groupValue: selectedType,
-            onChanged: onChanged,
+        InkWell(
```
Comment (guyluz11) [inline lib/presentation/molecules/study_type_selection_molecule.dart:43]: Please create a new radio button function that will include the text as well.
Also make sure that clicking on InkWell area also checks the radio button button and not just calling the method.

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by M4dhav** [comment]
All requested changes have been made. 
Please note that the radio button was already being checked on clicking InkWell but to make this effect more apparent I have added a delay of 200ms to navigation of the next page

File: lib/presentation/molecules/study_type_selection_molecule.dart [hunk 2]
```diff
@@ -49,26 +49,28 @@ class _StudyTypeSelectionMoleculeState
...
-            subtitle: 'recommended_evening',
-          ),
+          studyTypeRadioButton(onChanged, selectedType, TipType.analytical),
+          studyTypeRadioButton(onChanged, selectedType, TipType.creative),
         ],
       ),
     );
   }
+
+  Widget studyTypeRadioButton(void Function(TipType?)? onChanged,
```
Comment (guyluz11) [inline lib/presentation/molecules/study_type_selection_molecule.dart:59]: I think null is not necessary, we always transfer function and type
`void Function(TipType?)? `

File: lib/presentation/pages/intro_page.dart [hunk 3]
```diff
@@ -57,10 +57,11 @@ class _IntroPageState extends State<IntroPage> {
           PageViewModel(
             titleWidget: const SizedBox(),
             decoration: emptyPageDecoration(),
-            bodyWidget: StudyTypeSelectionMolecule(() {
+            bodyWidget: StudyTypeSelectionMolecule(() async {
               setState(() {
                 studyType = StudyTypeAbstract.instance!.studyType.name;
               });
+              await Future.delayed(const Duration(milliseconds: 200));
```
Comment (guyluz11) [inline lib/presentation/pages/intro_page.dart:64]: This duration should be const for all use cases in page

**Review by guyluz11 — APPROVED** [review]
Will improve the code but some changes are still needed, let me know if you want to add them

File: lib/presentation/pages/intro_page.dart (hunk 3 above)
Comment (M4dhav) [reply lib/presentation/pages/intro_page.dart:64]: I will make both these changes, I am about to open another PR for issue #18 so I will include these changes in that as well if its fine?

File: lib/presentation/molecules/study_type_selection_molecule.dart (hunk 2 above)
Comment (M4dhav) [reply lib/presentation/molecules/study_type_selection_molecule.dart:59]: This fix is included in #51

File: lib/presentation/pages/intro_page.dart (hunk 3 above)
Comment (guyluz11) [reply lib/presentation/pages/intro_page.dart:64]: External (at class level) const Duration var, not inside each transition

**Pull request merged**